src/main/java/net/javadiscord/javabot/listener/PingableNameListener.java

```diff
@@ -0,0 +1,115 @@
+		log.info("Loaded {} Nouns!", nouns.size());
+		log.info("Loaded {} Adjectives!", adjectives.size());
+	}
+
+	@Override
+	public void onGuildMemberJoin(GuildMemberJoinEvent event) {
+		if (!isPingable(event.getUser().getName()) && !canBypassCheck(event.getMember())) {
+			String newName = generateRandomName();
+			event.getMember().modifyNickname(newName).queue();
+			event.getUser().openPrivateChannel()
+					.flatMap(channel -> channel.sendMessageFormat("Your nickname has been set to %s since your username's first three characters were deemed as not-pingable.", newName))
+					.queue();
+		}
+	}
+
+	@Override
+	public void onGuildMemberUpdateNickname(GuildMemberUpdateNicknameEvent event) {
+		if (!isPingable(event.getNewNickname()) && !isPingable(event.getUser().getName()) && !canBypassCheck(event.getMember())) {
+			String newName = generateRandomName();
+			event.getMember().modifyNickname(newName).queue();
+			event.getUser().openPrivateChannel()
+					.flatMap(channel -> channel.sendMessageFormat("Your nickname has been set to `%s` since both your user- and nickname's first three characters were deemed as not-pingable.", newName))
+					.queue();
+		}
+	}
+
+	/**
+	 * Checks if the given name's first three characters contain ASCII characters not between 32 and 126.
+	 * @param name The name to check.
+	 * @return True if first three characters contain invalid characters, False if not.
+	 */
+	private boolean isPingable(String name) {
+		if (name == null) return true;
+		char[] nameChars = name.toCharArray();
+		for (int i = 0; i < 2; i++) {
+			char c = nameChars[i];
+			if (c < 32 || c > 126) {
+				return false;
+			}
+		}
+		return true;
+	}
+
+	/**
+	 * Generates a random name consisting of a noun, an adjective and a number from 1 to 9999.
+	 * @return The generated name.
+	 */
+	private String generateRandomName() {
+		String noun = this.nouns.get(random.nextInt(nouns.size()));
+		String adjective = this.adjectives.get(random.nextInt(adjectives.size()));
+		int number = random.nextInt(10000);
+		return StringUtils.capitalize(adjective) + StringUtils.capitalize(noun) + number;
+	}
+
+	/**
+	 * Reads strings from the given URL to a {@link List}, removes words with a dash.
+	 * @param url The URL to read.
+	 * @return A {@link List} of Strings.
+	 */
+	private static List<String> readStrings(String url) {
+		List<String> list;
+		try(Scanner scan = new Scanner(new URL(url).openStream()).useDelimiter("\\A")) {
+			String response = scan.next();
+			list = Arrays.stream(response.split("\n")).collect(Collectors.toList());
+		} catch (IOException e) {
+			e.printStackTrace();
+			list = new ArrayList<>();
+		}
+
+		list.removeIf(word -> word.contains("-"));
+		return list;
+	}
+
+	/**
+	 * Checks if the given {@link Member} can bypass the name-check.
+	 * @param member The {@link Member} to check.
+	 * @return Whether the Member can bypass name-checks or not.
+	 */
+	private static boolean canBypassCheck(Member member) {
```

Comment:
Do we need that? To be honest, I think this should apply for everyone.

Reply:
Yeah, you're right ig

Reply:
oh wait, this was done since the bot cannot edit the nicknames of people above itself

Reply:
What about using `canInteract` instead?

Reply:
alright